### PR TITLE
Multiple pools

### PR DESCRIPTION
**Pull request by puzza007**
Ability to have multiple pools with different sizes and options. A good use case would be to have a pool with pipelining enabled and one without.

### CODE REVIEW

**Comment by ibnfirnas** [comment]
:+1:

**Comment by ibnfirnas** [comment]
Another benefit is isolating pipelined requests to some API subset from pipelined request failures in another API subset of the same host.

File: .travis.yml [hunk 1]
```diff
@@ -3,9 +3,9 @@ sudo: true
 language: erlang
 
 otp_release:
-  - 17.5
```
Comment (ibnfirnas) [inline .travis.yml]: https://github.com/puzza007/katipo/blob/867dfc28f87b10f499aaeb1735ab5e1d240b8a83/rebar.config#L5

Comment (puzza007) [inline .travis.yml]: Ta! :family:

File: src/katipo_pool.erl [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    ChildSpec = #{id => PoolName,
+                  start => {katipo_pool_sup, start_link, Args},
+                  restart => permanent,
+                  shutdown => 2000,
+                  type => supervisor,
+                  modules => [katipo_pool_sup]},
+
+    ok = gproc_pool:new(PoolName, round_robin, [{size, PoolSize}]),
+    _ = [gproc_pool:add_worker(PoolName, worker_name(PoolName, N))
+         || N <- lists:seq(1, PoolSize)],
```
Comment (ibnfirnas) [inline src/katipo_pool.erl]: :eyes:

Comment (puzza007) [inline src/katipo_pool.erl]:
```erlang
ok = lists:foreach(fun(N) ->
                       N = gproc_pool:add_worker(PoolName, worker_name(PoolName, N))
                   end,
                   lists:seq(1, PoolSize)),
```

?

Comment (ibnfirnas) [inline src/katipo_pool.erl]: :trollface:

**Comment by puzza007** [comment]
@zoldar FYI new pools interface

**Comment by zoldar** [comment]
@puzza007 Cool! I want to get back to contributing here but for now I'm swamped with work :(

**Comment by puzza007** [comment]
Spoko. Czekam ;)

**Pull request merged**